### ISSUE TITLE
Unit tests are failing when Postgres persistent connection is enabled 

### ISSUE THREAD

**Issue by somayajulu** [Codeception/Codeception#6150]
#### What are you trying to achieve?
Unit test should pass when db persistent connection is enabled 
#### What do you get instead?
Receiving error "[PDOException] There is already an active transaction" 

```bash
[PDOException] There is already an active transaction

```
> 
This is the database.php file db configuration in Laravel 
'connections' => [
        
        'cockroach' => [
            'driver' => 'pgsql',
            'host' => env('DB_HOST', 'xxxxx'),
            'port' => env('DB_PORT', 'xxxxx'),
            'database' => env('DB_DATABASE', 'xxxxx'),
            'username' => env('DB_USERNAME', 'xxxx'),
            'sslmode' => 'require',           
            'sslkey' => env('DB_SSLKEY', "xxxx"),
            'sslcert' => env('DB_SSLCERT', "xxxxx"),           
            'options' => array(
            PDO::ATTR_ERRMODE  => PDO::ERRMODE_EXCEPTION,
            PDO::ATTR_EMULATE_PREPARES => true,
            PDO::ATTR_PERSISTENT => true),

        ],

    ]
Mocking database query builder to test functions written in Laravel repository classes (holds database logic)
```php
   $this->builder = Mockery::mock('Illuminate\Database\Query\Builder');
        $this->db = DB::shouldReceive('commit')->andReturnNull()
            ->shouldReceive('rollback')->andReturnNull()
            ->shouldReceive('beginTransaction')->andReturnNull()
            ->shouldReceive('getConnections')->andReturn([])
            ->shouldReceive('table')->andReturn($this->builder)
            ->shouldReceive('raw')->andReturn("");
```
### Details

* Codeception version: 4.1.17
* PHP Version: 7.4.4
* Operating System: Ubuntu
* Installation type: Composer
* List of installed packages (`composer show`)
* Suite configuration:

```yml

unit.suit.yml
actor: UnitTester
modules:
    enabled:
        - Asserts
        - \Helper\Unit
        - Laravel5
    step_decorators: ~
```

**Comment by Naktibalda** [comment]
All you gave here is a single line description in subject. It isn't a useful bug report or even support request.

Please provide exact information about how your tests are failing - if your database throws its own exception after catching database exception, make your best effort to extract information about database error.

Please provide configuration of your unit suite (it is important to know what modules are enabled and what settings are used)
and example of test code that fails, example of relevant application code may be useful too.

If the tests are using database, they aren't really unit tests, but this is definition problem, not test problem.
It is useful to separate tests that require database connectivity from tests that don't to separate test suites to improve performance.

**Comment by somayajulu** [comment]
@Naktibalda, Hi Naktibalda, I have updated the post with all details.
My unit test were failing when I enable 'persistent' = 'ture'in laravel's database.php file

**Comment by Naktibalda** [comment]
Have you got `database.default` setting in your Laracel configuration?

**Comment by somayajulu** [comment]
yes, its set 'default' => env('DB_CONNECTION', 'cockroach'),

**Comment by Naktibalda** [comment]
Please run codeception with `-vv` parameter and check if you see the following lines in the output:
```
[Database] Transaction started
``` 
```
[Database] Transaction cancelled; all changes reverted.
```

**Comment by somayajulu** [comment]
Yes I have run tests in debug mode by adding additional param '--debug' to the deception run command and i could see transaction   messages

for for test case its showing  [Database] Transaction started

and when it comes to other test its giving
[Database] Transaction cancelled; all changes reverted.

**Comment by somayajulu** [comment]
> Yes I have run tests in debug mode by adding additional param '--debug' to the deception run command and i could see transaction messages
> 
> for for test case its showing 

[Database] Transaction started
> 
> and when it comes to other test its giving

> [Database] Transaction cancelled; all changes reverted.

**Comment by somayajulu** [comment]
> > Yes I have run tests in debug mode by adding additional param '--debug' to the deception run command and i could see transaction messages
> > for for test case its showing
> 
> [Database] Transaction started
> 
> > and when it comes to other test its giving
> 
> [Database] Transaction cancelled; all changes reverted.

**Comment by somayajulu** [comment]
> > > Yes I have run tests in debug mode by adding additional param '--debug' to the deception run command and i could see transaction messages
> > > for for test case its showing
> > 
> > 
> > [Database] Transaction started
> > > and when it comes to other test its giving
> > 
> > 
> > [Database] Transaction cancelled; all changes reverted.

**Comment by Naktibalda** [comment]
Laravel module starts transaction before each test and rolls it back after each test.
This may cause issues if your application is using its own transactions.
If you don't want this automatic cleanup to happen you can set `cleanup: false` in parameters of Laravel module.

Laravel module attempts to close all database connections after each test.

I think that you will have to come up with a way to disable persistent connections in test environment.

**Comment by somayajulu** [comment]
I have tried giving `cleanup: false ` param under laravel5 module configuration but no luck.

Why does unit test requires a database connection actually we dont run any test directly on a db. 

Could you please suggest if there is a way to bypass checking db connection for unit test or a way to pass a test database connection details for database.

Regarding disabling persistent connection, yes i need to check for a workaround (where to provide it)

**Comment by Naktibalda** [comment]
> Why does unit test requires a database connection actually we dont run any test directly on a db.

Because Laravel module is enabled, do you really need it in unit tests?

**Comment by somayajulu** [comment]
Yeah that is required otherwise all test are failing.
